### PR TITLE
Port Greg's peaknd work

### PR DESCRIPTION
**Pull request by bbimber**

### CODE REVIEW

File: R/BFF_Demux.R [hunk 1]
```diff
@@ -257,129 +257,16 @@ GenerateCellHashCallsBFF <- function(barcodeMatrix, assay = "HTO", min_average_r
...
-      return(df)
-    } else if (calling_paramset == "bff_q19"){
-      SummarizeHashingCalls(seuratObj, label = 'bff_q19', columnSuffix = 'bff_q19', assay = assay, doHeatmap = TRUE)
-      df <- data.frame(cellbarcode = as.factor(colnames(seuratObj)), method = 'bff_q19', classification = seuratObj$classification.bff_q19, classification.global = seuratObj$classification.global.bff_q19, stringsAsFactors = FALSE)
-      return(df)
-    } else if (calling_paramset == "bff_q20"){
-      SummarizeHashingCalls(seuratObj, label = 'bff_q20', columnSuffix = 'bff_q20', assay = assay, doHeatmap = TRUE)
-      df <- data.frame(cellbarcode = as.factor(colnames(seuratObj)), method = 'bff_q20', classification = seuratObj$classification.bff_q20, classification.global = seuratObj$classification.global.bff_q20, stringsAsFactors = FALSE)
+    seuratObj <- BFFDemux(seuratObj = seuratObj, assay = assay, simple_threshold = as.logical(simple_threshold), doublet_thresh = doublet_thresh, neg_thresh = neg_thresh, pos_dist = pos_dist, neg_dist = neg_dist)
```
Comment (bbimber) [inline R/BFF_Demux.R:260]: why does this need as.logical()? cant we just assume we're given a boolean?

File: R/BFF_Demux.R [hunk 2]
```diff
@@ -389,203 +276,18 @@ GenerateCellHashCallsBFF <- function(barcodeMatrix, assay = "HTO", min_average_r
 }
 
 
-BFFDemux <- function(seuratObj, assay, calling_paramset) {
+BFFDemux <- function(seuratObj, assay, simple_threshold, doublet_thresh, neg_thresh, pos_dist, neg_dist) {
```
Comment (bbimber) [inline R/BFF_Demux.R:279]: my R is not necessarily good enough to know the difference, but I tend to like named args in this scenario. it's more verbose, but "simple_threshold = simple_threshold, doublet_thresh = doublet_thresh", etc. is probably less likely to break in a hard to detect manner if you ever change the downstream code.

File: R/BFF_Demux.R [hunk 3]
```diff
@@ -611,156 +313,87 @@ BFFDemux <- function(seuratObj, assay, calling_paramset) {
     print(paste0(cutoff, ': ', cutoffs[[cutoff]]))
   }
 
-  if (recover == FALSE) {
-    seuratObj <- .AssignCallsToMatrix(seuratObj, discrete, suffix = 'bff', assay = assay)
+  if (as.logical(simple_threshold) == TRUE) {
```
Comment (bbimber) [inline R/BFF_Demux.R:316]: same comment as above about as.logical()

**Review by bbimber — COMMENTED** [review]
i think this is good, but see some minor comments

File: R/BFF_Demux.R (hunk 1 above)
Comment (gboggy) [reply R/BFF_Demux.R:260]: Yes.  Thank you for pointing this out.  This was something I tried while debugging and didn't remove.

**Pull request merged**